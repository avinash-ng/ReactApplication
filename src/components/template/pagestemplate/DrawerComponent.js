import React, {useState} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import CssBaseline from '@material-ui/core/CssBaseline';
import AppBar from '@material-ui/core/AppBar';
import ToolBar from '../../molecules/Toolbar/ToolBar';
import List from '@material-ui/core/List';
// import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
// import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
// import InboxIcon from '@material-ui/icons/MoveToInbox';
// import MailIcon from '@material-ui/icons/Mail';
// import ReactDOM from 'react-dom';
import { useAuth0 } from "../../../react-auth0-spa"
import Button from "@material-ui/core/Button"
import history from "../../../utils/history"
// import Title from "../../atoms/Title/Title"
import sinon from 'sinon'
import './DrawerComponent.css'
import MainListItems from '../../molecules/List/MainListItems';
import ListWithSubMenu from '../../organisms/ListWithSubMenu/ListWithSubMenu';
const drawerWidth = 240;




const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    height: 'inherit'
  },
  appBar: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  // necessary for content to be below app bar
  toolbar: theme.mixins.toolbar,
  content: {
    marginTop: '64px',
    flexGrow: 1,
    height: 'inerit',
    backgroundColor: theme.palette.background.default,
    padding: theme.spacing(3),
  },
}));

const clicked = (text) => {
  var loweredText = "/" + text.toLowerCase();

  history.push(loweredText);
  // ReactDOM.render(, document.getElementById("content"));
}





export default function PermanentDrawerLeft({ children }) {
  
  const classes = useStyles();
  //hardcode user name
  
  // var userName = user===undefined? 'avinash': user.name
  const { user, logout, isAuthenticated } = useAuth0();
  const [anchorEl, setAnchorEl] = useState(null)

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  }
  const hoverHandler = () => {
    document.getElementsByClassName('list-div')[0].style.display = ''
  }
  const leaveHandler = () => {
      document.getElementsByClassName('list-div')[0].style.display = 'none'
  }


    return (
       
      <div className={classes.root} >
      <CssBaseline />
      <AppBar position="fixed" className={classes.appBar}>
          <ToolBar style={{ marginLeft: '1100px', width: '75px', backgroundColor: 'white' }} name={user.name} buttonName="logout" variant="contained" onClick={logout}/>
      </AppBar>
      <Drawer
        className={classes.drawer}
        variant="permanent"
        classes={{
          paper: classes.drawerPaper,
        }}
        anchor="left"
      >
        <div className={classes.toolbar} />
        <Divider />
          <List>
           
          {/* {['Home', 'About', 'Contact', 'Gallery'].map((text, index) => (
            <ListItem button style={{marginLeft: '0px'}} key={text} onClick={() => clicked(text)}>
              <ListItemText primary={text} />
            </ListItem>
          ))} */}
            
              <MainListItems clicked={clicked}/>
              {/* <div className="dropdown">
              <ListItem button key='Image' className="dropbtn">
                <ListItemText primary={'Image'} />
                </ListItem>
                <div className="dropdown-content">
                  <a href="/edit">Edit</a>
                  <a href="/delete">Delete</a>
                  <a href="/upload">Upload</a>
                  </div>
                </div>
              */}
            <ListWithSubMenu />
              </List> 
          

        <Divider />
          
        {/* <div className="list-div">
            <ul>
              <li>
                <Button> Edit </Button>
              </li>
              <li>
                <Button> Delete </Button>
              </li>
              <li>
                <Button>  </Button>
              </li>
            </ul>    
        </div> */}
          
          
        
      </Drawer>
      <main className={classes.content} id="main">
        {/* <div className={classes.toolbar} /> */}
        
          <div id="content"> {children}</div>
      </main>
    </div>
  );
}
