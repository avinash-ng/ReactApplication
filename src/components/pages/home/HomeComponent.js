import React from 'react';
import DrawerComponent from "../../template/pagestemplate/DrawerComponent"
export const ContentComponent = () => {
    return (
        <div>
            <div className="heading">
                <h1>Welcome to home page</h1>
            </div>
            <div className="content">
                <span style={{fontSize: '25px'}}>In this simple app i have provided some functionalities like authentication with auth0.
                I have used react as a front end and nodejs backend.
                And for sending email i used a dependency named nodemailer in the backend.
                I also stored images uploaded by the end user in the database with the help of another node dependency named multer. It is used for handling form data like images, files.
                For this applications the pattern followed is atomic design pattern, where the main elements are atoms, molecules, organisms, pages, templates.
                </span>
            </div>
        </div>
    )
}
const HomeComponent = () => {

    return (

        <div>
            <DrawerComponent> <ContentComponent/> </DrawerComponent>

        </div>
    )
}
export default HomeComponent;