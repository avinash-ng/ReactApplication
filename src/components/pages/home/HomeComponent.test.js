const { shallow, configure, mount } = require("enzyme")
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import { useAuth0 } from '../../../react-auth0-spa'
import DrawerComponent from "../../template/pagestemplate/DrawerComponent";
import { ContentComponent } from './HomeComponent'
import HomeComponent from "./HomeComponent";
import { render } from '@testing-library/react';
configure({ adapter: new Adapter() });
const user = {
    email: "av@gmail.com",
    email_verified: true,
    sub: "google-oauth2|12345678901234"
}
jest.mock("../../../react-auth0-spa");
describe('testing <HomeComponent /> ', () => {
    beforeEach(() => {
        // Mock the Auth0 hook and make it return a logged in state
        useAuth0.mockReturnValue({
            isAuthenticated: true,
            user,
            logout: jest.fn(),
            loginWithRedirect: jest.fn()
        });
    });

    it('renders <DrawerComponent />  correctly', () => {

        const RenderComponent = mount(<HomeComponent> </HomeComponent>)
        expect(RenderComponent.find(DrawerComponent).exists()).toBe(true)
    })

    it('renders <ContentConponent />  correctly', () => {

        const RenderComponent = mount(<HomeComponent> </HomeComponent>)
        expect(RenderComponent.find(ContentComponent).exists()).toBe(true)
    })

    it('render test properly', () => {
        const { getByText } = render(<HomeComponent> </HomeComponent>)
        expect(getByText('Welcome to home page')).toBeInTheDocument()
    })
    
    it('matches snapshot', () => {
        const RenderComponent = shallow(<HomeComponent ></HomeComponent>)
        expect(RenderComponent).toMatchSnapshot()
    })
})