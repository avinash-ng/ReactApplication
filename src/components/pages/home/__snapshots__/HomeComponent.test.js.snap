// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <HomeComponent />  matches snapshot 1`] = `ShallowWrapper {}`;
